app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -257,6 +264,37 @@ class NavigationOverlayFragment : Fragment() {
         activity!!.moveTaskToBack(true)
     }
 
+    private fun observeAccountState(): Disposable {
+        val fxaRepo = serviceLocator.fxaRepo
+
+        return fxaRepo.accountState.subscribe { accountState ->
+            when (accountState) {
+                is AccountState.AuthenticatedWithProfile -> {
+                    val fxaProfile = accountState.profile
+                    if (fxaProfile.avatar != null) {
```

Comment:
What happens when the profile avatar is null? We need to set an image, right?

My assumption is that, if the states transition how we expect them to, we'll still be using the image from the last state (authenticated_no_profile) so, in practice, it'll look fine to users in the current implementation. However, this logic couples the previous state to the current state, making you consider all states the code can be in instead of just this one state when making changes to this state.

Instead, we should act like this state exists in isolation and handle all of its cases: i.e. we should set an image if `fxaProfile.avatar` is not null or if it's null, even if the value we set is redundant to the previous state. That way, we don't have to consider the previous state at all: the amount of code we need to consider when making changes to this code is reduced.

In general, Rx (and functional programming) want you to consider functions/states in isolation (including this one!) to reduce the amount of code/state you need to consider at one time.

---

That being said, it looks like Severin is updating the `FxaProfile` to return an `ImageSetStrategy` instead of an avatar URL https://github.com/mozilla-mobile/firefox-tv/pull/2723/files#diff-6b880929405068b06513952ce832061eR169 that already encapsulates the logic to convert null URLs to default avatars and such. **Once you rebase this code, I think this explicitly setting a value for avatar == null will be taken care of.** I left the explanation above in to hopefully provide more context on some Rx best practices!